Only put overlay on link, don't modify file content.

ringawho opened the pull request:
Thank you for developing this package. I really enjoy it.

The current implementation of displaying sliced images directly modifies the file content, which may lead to frequently auto save for me.

I have modified its implementation so that image display no longer alters the file content. Instead, it achieves this by putting overlay on link. However, one drawback is that the image slices may exceed the default font height.

jcfk commented:
Thank you for your contribution. This is an interesting solution to a legitimate problem. My thoughts:

- The image is going to be navigated as a single line in certain contexts (eg. evil-mode), which could be confusing.
- Can you explain the front-sliced logic to me? If one wanted slices to be as uniform as possible (which I think is a good idea), wouldn't you just make them uniform.

Regarding org-sliced-images-mode, I would make it global.

Because this is a major change, I'll also wait to see whether there are any other comments.

ringawho commented:
Thanks for your advice. 

1. In most cases, the link supports display sliced image that are the same height as one line. However, if the link is too short or the image is too large, the sliced image may be taller than one line. To address this, I add a new custom variable `org-sliced-images-max-sliced-height`. When sliced image is too taller, the image will be zoom out.
2. I've ensured uniformity in the sliced images. Initially, I split the image unevenly to make it a multiple of the row height, but I realized that this approach wasn't necessary.
3. Additionally, org-sliced-images-mode should be a global mode, so I've added `:global t`.
